[dendendenden04] iP

dendendenden04 opened the pull request.

dendendenden04 committed:
Level-0

Changed bot name to Psyduck

dendendenden04 committed:
Level - 01

Added echo function to Psyduck bot.
Use command "Bye" to exit the bot.

dendendenden04 committed:
Level - 02

Added a list function for Psyduck.
Any text other than the special commands will add the task to a list.
Added exit as an option to exit the program.

dendendenden04 committed:
Level - 2.5

Unfinished debugging to do with marking and unmarking tasks.

dendendenden04 committed:
Reworked Psyduck.java to process inputs better

TodoList.java: added ability to get task description

dendendenden04 committed:
Renamed ToDoList class to ToDo class

Refactor code to fit inheritance for the ToDo class from the Task class

Currently there are errors with method calls for markDone(), unmarkDone() and Todo()

dendendenden04 committed:
Task classes: Changed properties of Task class to accommodate child
classes

ToDo class now works as a child of the Task class

Prior errors are now fixed

dendendenden04 committed:
Add Deadline.java

Deadline class inherits from Task class with a deadline property

Adjust Psyduck main() method to accommodate the new Deadline class

Psyduck can now add tasks that are of Deadline type

dendendenden04 committed:
Add Event.java

Contains the Event class which inherits from the Task class.

Add addEvent() method to Psyduck class

To accommodate the new task type

dendendenden04 committed:
Delete ToDoList.java

Outdated ToDo class

dendendenden04 committed:
Refactor code in Psyduck.java

Refactor code to meet code quality standards

dendendenden04 committed:
Merge branch 'master' of https://github.com/dendendenden04/ip

dendendenden04 committed:
Refactor Psyduck class by auto indenting

adenteo commented on src/main/java/Psyduck.java:
```diff
@@ -0,0 +1,184 @@
+        int taskNum;
+        String input = in.nextLine();
+        input = input.trim();
+        int space = input.indexOf(" ");
+        String command;
+        if (space == -1) {
+            command = input;
+        } else {
+            command = input.substring(0, space);
+        }
```
Perhaps it is better to modify this into a Boolean function to check whether is it a single/multi-word user input?

dendendenden04 replied:
Good idea, thank you for your suggestion.

adenteo commented on src/main/java/Psyduck.java:
```diff
@@ -0,0 +1,184 @@
+        default:
+            handleInvalidCommand();
+        }
+    }
```
Good use of switch statement for a clean description of each case!

adenteo commented on src/main/java/Psyduck.java:
```diff
@@ -0,0 +1,184 @@
+            return;
+        }
+        int byPos = input.indexOf("/by");
+        if (byPos == -1) {
```
Perhaps it is better to use Boolean functions for these code segments on input validation? This is applicable to a few other segments as well that I have noted out.

adenteo commented on src/main/java/Psyduck.java:
```diff
@@ -0,0 +1,184 @@
+        }
+        String description = input.substring(space + 1, fromPos);
+        String from = input.substring(fromPos + 6, toPos);
+        String to = input.substring(toPos + 4);
```
Perhaps it is better to avoid magic numbers to make it clearer to the reader? Could consider using named constants or other methods to split the user input.

adenteo reviewed (commented):
Overall, I found your code relatively easy to read. Every function was simple and neat and the flow of events were quite clear. I noted a few segments regarding your validation of user inputs which could be quite confusing. There was also a segment where you used magic numbers to obtain substrings of the user input, which may not be intuitive.  Consider using named constants or creating functions for input validation?

dendendenden04 committed:
Reorganise code to fit command related methods into Command.java

Most of the code that do not have to deal with managing the list of
task is reorganised into a Command class

dendendenden04 committed:
Add NullPointerException catch for marking and unmarking tasks

This will catch errors to prevent users from accessing unintialised
Task.

Add a line to remind users how many tasks are in the list

dendendenden04 committed:
Add a comment to branch

dendendenden04 committed:
Add NumberFormatException for marking/unmarking of tasks

This exception will catch erroneous user inputs who input a string
after the command "mark". Also minor changes to the formatting for
printing of total number of tasks.

dendendenden04 committed:
Merge branch 'master' into branch-Level-5

dendendenden04 committed:
Update create<Task> methods

Now correctly outputs 'task' or 'tasks' depending on number of
tasks in the list.

dendendenden04 committed:
Minor edit in spacing between methods

dendendenden04 committed:
Merge branch 'branch-Level-5'

dendendenden04 committed:
Change tasks from array type to ArrayList<> type

Also done respective changes to each function that is affected.
These functions perform the same but supports the new data type

Add remove functionality for the list

dendendenden04 committed:
Incomplete commit

dendendenden04 committed:
merging branch back

dendendenden04 committed:
Merge branch 'branch-Level-6'

dendendenden04 committed:
Add PsyduckExceptions

dendendenden04 committed:
Add file reading and writing capability

Add .gitignore for 'save.txt'

'save.txt' is the file created and read from for saving purposes

dendendenden04 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Psyduck.java

dendendenden04 committed:
Remove lines generated by resolving merge conflict

okkhoy commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,210 @@
+    }
+
+    public static void createToDo(String input, int space) {
+        if (space == NOT_FOUND) {
+            linePrint();
+            System.out.println("Todo task cannot be empty! >:(");
+            linePrint();
+            return;
+        }
```
Can be handled as an exception

okkhoy commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,210 @@
+        }
+        switch (command) {
+        case "bye":
+            //fallthrough
```
👍

okkhoy commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,210 @@
+    }
+
+    public static void processCommands() {
+        String input = in.nextLine();
+        input = input.trim();
+        int space = input.indexOf(" ");
```
Can be extracted to a separate method that does parsing

okkhoy commented on src/main/java/Psyduck.java:
```diff
@@ -0,0 +1,102 @@
+        if (taskCount == 0) { //list is empty
+            System.out.println("List is empty.");
+        } else {
+            for (int i = 0; i < taskCount; i++) {
+                System.out.print(Integer.toString(i + 1) + ".");
+                System.out.println(tasks.get(i));
+            }
```
Make happy path prominent

okkhoy commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,86 @@
+            File f = new File(filepath);
+            if (f.createNewFile()) {
+                System.out.println("File created: " + f.getName());
+            } else {
+            }
```
please do not have empty catch blocks. if you choose, you may just check for the condition in a guard clause

okkhoy reviewed (commented):
1. The codebase is not well organized; please complete packages task soon.
2. There is no exception handling yet. 
3. Please include Javadoc comments for all non-trivial methods and classes.

dendendenden04 committed:
Refactor code to make it more OOP

dendendenden04 committed:
Add Level-9 Find functionality

dendendenden04 committed:
Merge pull request #1 from dendendenden04/branch-Level-9

Add Level-9 Find functionality

dendendenden04 committed:
Added JavaDoc comments to most public classes/methods

dendendenden04 committed:
Merge pull request #2 from dendendenden04/branch-A-JavaDoc

Added JavaDoc comments to most public classes/methods

dendendenden04 committed:
Edit README.md

dendendenden04 committed:
Merge branch 'master' of https://github.com/dendendenden04/ip

dendendenden04 committed:
Update README.md for user guide

Minor changes to Ui.java for line printing optimisation

dendendenden04 committed:
Fix errors with saving text file

Updated README to reflect instructions on opening command terminal.

dendendenden04 committed:
Refactor code to reduce SLOC

Edit minor error to README.md, mainly task description was not
capitalised properly.
